src/PSRule.Rules.Azure/rules/Azure.AKS.Rule.ps1

```diff
@@ -245,6 +245,83 @@ Rule 'Azure.AKS.AuditLogs' -Type 'Microsoft.ContainerService/managedClusters' -T
     }
 }
 
+# Synopsis: AKS clusters should collect platform diagnostic logs to monitor the state of workloads.
+Rule 'Azure.AKS.PlatformLogs' -Type 'Microsoft.ContainerService/managedClusters' -Tag @{ release = 'GA'; ruleSet = '2021_09'; } {
+    $configurationLogCategoriesList = $Configuration.GetStringValues('AZURE_AKS_ENABLED_PLATFORM_LOG_CATEGORIES_LIST');
+
+    if ($configurationLogCategoriesList.Length -eq 0) {
+        return $Assert.Pass();
+    }
+
+    $diagnosticLogs = @(GetSubResources -ResourceType 'Microsoft.Insights/diagnosticSettings', 'Microsoft.ContainerService/managedClusters/providers/diagnosticSettings');
+
+    $Assert.Greater($diagnosticLogs, '.', 0);
+
+    $availableLogCategories = @{
+        Logs = @(
+            'cluster-autoscaler', 
+            'kube-apiserver', 
+            'kube-controller-manager', 
+            'kube-scheduler'
+        )
+        Metrics = @(
+            'AllMetrics'
+        )
+    }
+
+    $configurationLogCategories = @($configurationLogCategoriesList | Where-Object {
+        $_ -in $availableLogCategories.Logs
+    });
+
+    $configurationMetricCategories = @($configurationLogCategoriesList | Where-Object {
+        $_ -in $availableLogCategories.Metrics
+    });
+
+    $logCategoriesNeeded = [System.Math]::Min(
+        $configurationLogCategories.Length, 
+        $availableLogCategories.Logs.Length
+    );
+
+    $metricCategoriesNeeded = [System.Math]::Min(
+        $configurationMetricCategories.Length, 
+        $availableLogCategories.Metrics.Length
+    );
+
+    $logCategoriesJoinedString = $configurationLogCategoriesList -join ', ';
+
+    foreach ($setting in $diagnosticLogs) {
+        $platformLogs = @($setting.Properties.logs | Where-Object {
+            $_.enabled -and
+            $_.category -in $configurationLogCategories -and
+            $_.category -in $availableLogCategories.Logs
+        });
+
+        $metricLogs = @($setting.Properties.metrics | Where-Object {
+            $_.enabled -and 
+            $_.category -in $configurationMetricCategories -and
+            $_.category -in $availableLogCategories.Metrics
+        });
+
+        $platformLogsEnabled = ($null -ne $platformLogs -and $platformLogs.Length -eq $logCategoriesNeeded) -and 
```

Comment:
I didn't check but can I use an Assertion here? Wasn't sure which one I can use.

Reply:
Technically you could use `$Assert.HasFieldValue` but like this `$Assert.HasFieldValue($platformLogs, 'Length', $logCategoriesNeeded)` or `$Assert.HasFieldValue($platformLogs.Length, '.', $logCategoriesNeeded)`.

Reply:
Thanks @BernieWhite. I've updated to use `HasFieldValue`.